HeterogeneousCore/AlpakaInterface/interface/atomicMaxPair.h

```diff
@@ -0,0 +1,38 @@
+#ifndef HeterogeneousCore_AlpakaCore_interface_atomicMaxPair_h
+#define HeterogeneousCore_AlpakaCore_interface_atomicMaxPair_h
+#include <alpaka/alpaka.hpp>
+
+#include "FWCore/Utilities/interface/bit_cast.h"
+#include "HeterogeneousCore/AlpakaInterface/interface/config.h"
+
+// Note: Does not compile with ALPAKA_FN_ACC on ROCm
+template <typename TAcc, typename = std::enable_if_t<alpaka::isAccelerator<TAcc>>, typename F>
+ALPAKA_FN_HOST_ACC ALPAKA_FN_INLINE void atomicMaxPair(const TAcc& acc,
+                                                       unsigned long long int* address,
+                                                       std::pair<unsigned int, float> value,
+                                                       F comparator) {
+#if defined(__CUDA_ARCH__) or defined(__HIP_DEVICE_COMPILE__)
+  unsigned long long int val = (static_cast<unsigned long long int>(value.first) << 32) + __float_as_uint(value.second);
+  unsigned long long int ret = *address;
+  while (comparator(value,
+                    std::pair<unsigned int, float>{static_cast<unsigned int>(ret >> 32 & 0xffffffff),
+                                                   __uint_as_float(ret & 0xffffffff)})) {
+    unsigned long long int old = ret;
+    if ((ret = atomicCAS(address, old, val)) == old)
+      break;
+  }
+#else
+  unsigned long long int val =
+      (static_cast<unsigned long long int>(value.first) << 32) + edm::bit_cast<unsigned int>(value.second);
+  unsigned long long int ret = *address;
+  while (comparator(value,
+                    std::pair{static_cast<unsigned int>(ret >> 32 & 0xffffffff),
+                              edm::bit_cast<float>(static_cast<unsigned int>(ret & 0xffffffff))})) {
+    unsigned long long int old = ret;
+    if ((ret = alpaka::atomicCas(acc, address, old, val)) == old)
+      break;
+  }
```

Comment:
For the CPU serial case (one thread, one block) we could use a simplified version:
```c++
  if (value.second > edm::bit_cast<float>(static_cast<unsigned int>(*address & 0xffffffff))) {
    unsigned long long int store = (static_cast<unsigned long long int>(value.first) << 32) + edm::bit_cast<uint32_t>(value.second);
    *address = store;
  }
```

Anyway, this gives identical results.

Reply:
However now I am _very_ confused...

I changed the code from `value.second > edm::bit_cast<...>` to `value.second < edm::bit_cast<...>`, and the HLT results _did not change_.

Reply:
I think I understand: this code is responsible for computing the "method 0" _time_ of the HCAL sample - which we likely do not use at HLT.